types/heatmap.js/heatmap.js-tests.ts

```diff
@@ -1,42 +1,283 @@
+// tslint:disable-next-line no-object-literal-type-assertion
+const container = {} as HTMLElement;
 
+// -- h337.HeatmapConfiguration --
 
-var baseLayer = L.tileLayer(
-    'http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
-        attribution: 'Map data &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors, <a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a href="http://cloudmade.com">CloudMade</a>',
-        maxZoom: 18
-     });
-
-var testData: HeatmapData = {
-    max: 8,
-    data: [
-        {
-            lat: 24.6408,
-            lng:46.7728,
-            count: 3
-        }, {
-            lat: 50.75,
-            lng: -1.55,
-            count: 1
+{
+    const config: h337.HeatmapConfiguration = { container };
+    config; // $ExpectType HeatmapConfiguration<"value", "x", "y">
+}
+
+{
+    // $ExpectError
+    const config: h337.HeatmapConfiguration = {};
+}
+
+{
+    const config: h337.HeatmapConfiguration = {
+        container,
+        xField: 'x',
+        yField: 'y',
+        valueField: 'value',
+    };
+}
+
+{
+    // $ExpectError
+    const config: h337.HeatmapConfiguration = {
+        container,
+        valueField: 'foo',
+    };
+}
+
+{
+    const config: h337.HeatmapConfiguration<'foo'> = {
+        container,
+        valueField: 'foo',
+    };
+    config; // $ExpectType HeatmapConfiguration<"foo", "x", "y">
+}
+
+// -- h337.create --
+
+{
+    h337.create(); // $ExpectError
```

Comment:
We don't really need a test that a required parameter is required.

Reply:
0e6cdf8acde1c2261e4ef3d25113b3905bf26e52